### PR TITLE
Hbire vector take2

### PR DESCRIPTION
**Pull request by amirroth**
This is the first working version of the vectorized HeatBalanceIntRadExchange. There are still a number of things that can be improved (e.g., sorting surfaces to make windows contiguous, experimenting with 32-bit floating-point, moving the hand-rolled padded 2D array to a real 2D array, a number of other things) but I wanted to get in there and get on with my life. Mike Witte reviewed and helped find the final bug. Thanks MJW! The remaining small diffs are in the 10th decimal place (another reason to change to single-precision floating-point).

### CODE REVIEW

**Comment by Myoldmopar** [comment]
@amirroth This does look good.  This _replaces_ the other [pull request](https://github.com/NREL/EnergyPlus/pull/4886) now, right?  If so, I'll close it and delete the branch over there.

I'll make a commit to clean up the compiler warning about 2 unused variables, and then we'll work to review it and get it in.  Thanks.

**Comment by amirroth** [comment]
Yes, I will delete the RowMajorWithVectorization branch as soon as this one
is in.

-Amir

**Comment by amirroth** [comment]
JDG put me through a code review at ASHRAE. He demanded documentation and a unit test, which I have now provided.

File: src/EnergyPlus/HeatBalanceIntRadExchange.cc [hunk 1]
```diff
@@ -247,142 +268,154 @@ namespace HeatBalanceIntRadExchange {
...
+			// Both __restrict and __assume_aligned hints require raw pointers, hence need to use raw pointers here.  These 
+			// pointers do not manage or "own" memory, they simply point at memory in a way that allows the compiler to optimize
+			// better
+			for ( int ZoneSurfNum = 0; ZoneSurfNum < zvfi.NumOfSurfaces; ++ZoneSurfNum ) {
+				vecSurfaceTempK4[ ZoneSurfNum ] = pow_4( vecSurfaceTempK4 [ ZoneSurfNum ] + KelvinConv );
+			} // for ZoneSurfNum
+
+
+			// See comments above for explanation of __restrict and __ep_assume_aligned
+			Real64 * __restrict vecNetLWRadToSurf_Temp( &NetLWRadToSurf_Temp[ 0 ] );
```
Comment (jasondegraw) [inline src/EnergyPlus/HeatBalanceIntRadExchange.cc:354]: @amirroth Is this line needed? clang is griping about unused variables, and it looks like the same thing is happening down on line 373. Similar question for the three lines after this one.

Comment (amirroth) [inline src/EnergyPlus/HeatBalanceIntRadExchange.cc:354]: Sorry, the loop below needs to be:

for ( int ZoneSurfNum = 0; ZoneSurfNum < zvfi.NumOfSurfaces; ++ZoneSurfNum
) {
        vecNetLWRadToSurf_Temp[ ZoneSurfNum ] = 0.0;
        vecIRfromParentZone_Temp[ ZoneSurfNum ] = 0.0;
} // ZoneSurfNum

Not sure why g++ didn't complain about this.

**Comment by Myoldmopar** [comment]
@amirroth Per our side conversations, I'll close this PR so that the idiot @nrel-bot stops hounding @lgentile every week, but I won't delete the branch.  Whenever we are ready to tackle this again, the changes will still be here.

**Comment by amirroth** [comment]
I'm not done with you yet, I'm just busy with other things.

**Pull request closed without merging**